pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -234,7 +235,7 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 					if discReq.ErrorDetail != nil {
 						adsLog.Warnf("ADS:CDS: ACK ERROR %v %s (%s) %v", peerAddr, con.ConID, con.node.ID, discReq.String())
 						errCode := codes.Code(discReq.ErrorDetail.Code)
-						incrementXDSRejects(cdsReject, con.node.ID, errCode.String())
+						incrementXDSRejects(cdsReject, con.node.ID, fmt.Sprintf("%s:%s", errCode.String(), discReq.ErrorDetail.GetMessage()))
```

Comment:
Does this create issues with Prometheus labels because of message size? If yes, I will close this PR as any way it is being logged above.

Reply:
@mandarjog ^^^?

Reply:
labels that are large or of unbounded cardinality should not be used.
Instead classify the messages and then add a messageClass label.

Reply:
@mandarjog Ok. Thank you.  I will remove it.
@rshriram  I also realized that from our logs, that this warning message with huge discovery request is actually spamming the logs. I am actually adding additional debug log that has the whole proto string print and convert the warning to just print the error detail + message.